src/gui/torrentoptionsdialog.cpp

```diff
@@ -76,11 +84,28 @@ TorrentOptionsDialog::TorrentOptionsDialog(QWidget *parent, const QVector<BitTor
     const bool isFirstTorrentDHTDisabled = torrents[0]->isDHTDisabled();
     const bool isFirstTorrentPEXDisabled = torrents[0]->isPEXDisabled();
     const bool isFirstTorrentLSDDisabled = torrents[0]->isLSDDisabled();
+    const bool isFirstTorrentLSequentialEnabled = torrents[0]->isSequentialDownload();
+    const bool isFirstTorrentFirstLastPiecesEnabled = torrents[0]->hasFirstLastPiecePriority();
 
     m_torrentIDs.reserve(torrents.size());
     for (const BitTorrent::Torrent *torrent : torrents)
     {
         m_torrentIDs << torrent->id();
+        if (allSameAutoTMM)
+        {
+            if (torrent->isAutoTMMEnabled() != isFirstTorrentAutoTMMEnabled)
```

Comment:
can't we simplify this code with something like 

```diff
diff --git a/src/gui/torrentoptionsdialog.cpp b/src/gui/torrentoptionsdialog.cpp
index 49ac52cf7..09eddc0af 100644
--- a/src/gui/torrentoptionsdialog.cpp
+++ b/src/gui/torrentoptionsdialog.cpp
@@ -53,6 +53,24 @@ namespace
             slider->setMaximum(value);
         slider->setValue(value);
     }
+
+    template<typename C, typename F>
+    bool allSame(const C &container, const F &unary)
+    {
+        auto b = std::begin(container);
+        const auto e = std::end(container);
+
+        if (b == e) return true;
+
+        const auto v = std::invoke(unary, *b++);
+        while (b != e)
+        {
+            if (v != std::invoke(unary, *b++))
+                return false;
+        }
+
+        return true;
+    }
 }

 TorrentOptionsDialog::TorrentOptionsDialog(QWidget *parent, const QVector<BitTorrent::Torrent *> &torrents)
@@ -64,7 +82,7 @@ TorrentOptionsDialog::TorrentOptionsDialog(QWidget *parent, const QVector<BitTor

     Q_ASSERT(!torrents.empty());
     const auto *session = BitTorrent::Session::instance();
-    bool allSameUpLimit = true, allSameDownLimit = true, allSameRatio = true, allSameSeedingTime = true
+    bool allSameUpLimit = allSame(torrents, &BitTorrent::Torrent::downloadLimit), allSameDownLimit = true, allSameRatio = true, allSameSeedingTime = true
             , allTorrentsArePrivate = true, allSameDHT = true, allSamePEX = true,  allSameLSD = true;

     const int firstTorrentUpLimit = qMax(0, torrents[0]->uploadLimit());
```

maybe also see `std::equal`

Reply:
>can't we simplify this code with something like

Maybe it's simplified for those who can read that :). I'm not familiar with std and i wouldn't know how to change/fix something like that (if asked to).